R/diagnose_design.R

```diff
@@ -103,37 +103,41 @@ diagnose_design <- function(...,
     
     check_design_class(designs)
   }
-
-  # simulate if needed ------------------------------------------------------
-
+  
+  
   if (!exists("simulations_df")) {
+    # simulate if needed ------------------------------------------------------
     simulations_df <- simulate_design(!!!dots_quos, sims = sims)
+    diagnosands <- lapply(designs, function(x) 
+      diagnosands %||% attr(x, "diagnosands") %||% default_diagnosands)
```

Comment:
hmm not as clean as I thought

Reply:
```
diagnosands %||% lapply(designs, ...)
```

then down below, mapply should vectorize a single function  or list correctly - but needs to be verified.

Reply:
seems to work great